service/controller/clusterapi/v29/cluster_resource_set.go

```diff
@@ -595,37 +611,8 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 	}
 
 	initCtxFunc := func(ctx context.Context, obj interface{}) (context.Context, error) {
-		cr, err := key.ToCluster(obj)
-		if err != nil {
-			return nil, microerror.Mask(err)
-		}
-
-		var tenantClusterAWSClients aws.Clients
-		{
-			arn, err := credential.GetARN(config.K8sClient, cr)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
-			c := config.HostAWSConfig
-			c.RoleARN = arn
-
-			tenantClusterAWSClients, err = aws.NewClients(c)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
-		}
-
-		c := controllercontext.Context{
-			Client: controllercontext.ContextClient{
-				ControlPlane: controllercontext.ContextClientControlPlane{
-					AWS: config.ControlPlaneAWSClients,
-				},
-				TenantCluster: controllercontext.ContextClientTenantCluster{
-					AWS: tenantClusterAWSClients,
-				},
-			},
-		}
-		ctx = controllercontext.NewContext(ctx, c)
+		cc := controllercontext.Context{}
+		ctx = controllercontext.NewContext(ctx, cc)
```

Comment:
I wanted to fix this for a while now. This is the only thing that should happen here. The rest should be done in resource implementations.